pkg/auth/handler/saml/login_finish.go

```diff
@@ -20,9 +26,71 @@ func NewLoginFinishHandlerLogger(lf *log.Factory) *LoginFinishHandlerLogger {
 }
 
 type LoginFinishHandler struct {
-	Logger *LoginFinishHandlerLogger
+	Logger                     *LoginFinishHandlerLogger
+	SAMLService                HandlerSAMLService
+	SAMLSessionService         SAMLSessionService
+	AuthenticationInfoResolver SAMLAuthenticationInfoResolver
+	AuthenticationInfoService  SAMLAuthenticationInfoService
 }
 
 func (h *LoginFinishHandler) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
+	authInfoID, ok := h.AuthenticationInfoResolver.GetAuthenticationInfoID(r)
+	if !ok {
+		h.Logger.Warningln("authentication info id is missing")
+		http.NotFound(rw, r)
```

Comment:
Returning 404 does not seem appropriate here. I think we do not expect people to invoke this endpoint directly. Maybe `http.Error(w, "invoking this endpoint directly is not supported", http.StatusBadRequest)`?